client/src/modules/stock/entry/entry.js

```diff
@@ -341,8 +359,19 @@ function StockEntryController(
     if (!vm.movement.entry_type) {
       return Notify.danger('ERRORS.ER_NO_STOCK_SOURCE');
     }
+    vm.$loading = true;
     mapEntry.form = form;
-    return mapEntry[vm.movement.entry_type].submit();
+    return mapEntry[vm.movement.entry_type].submit()
+      .then(toggleLoadingIndicator)
+      .finally(forceReload);
+  }
+
+  function toggleLoadingIndicator() {
+    vm.$loading = !vm.$loading;
+  }
+
+  function forceReload() {
+    $state.reload();
```

Comment:
Oof.  Why do we need to force reload?  Why not just `form.$setPristine()` and delete the state data?